### PR TITLE
allow custom txGas for trusted forwarder+sd k frontend (handle sendTxn

### PR DESCRIPTION
**Pull request by livingrock7**

### CODE REVIEW

File: src/Biconomy.js [hunk 1]
```diff
@@ -807,12 +809,14 @@ async function handleSendTransaction(engine, payload, end) {
           metaTxApproach = smartContractMetaTransactionMap[contractAddr];
         }
 
-        let gasLimit = payload.params[0].gas;
+        let gasLimit = payload.params[0].gas || payload.params[0].gasLimit;
+        let txGas = payload.params[0].txGas;
```
Comment (tomarsachin2271) [inline src/Biconomy.js:813]: why gasLimit was not used for this purpose?

File: src/Biconomy.js [hunk 2]
```diff
@@ -878,14 +881,14 @@ async function handleSendTransaction(engine, payload, end) {
...
+                txGas = await contract.estimateGas[methodSignature](...paramArrayForGasCalculation, { from: account });
                  // do not send this value in API call. only meant for txGas
-                gasLimitNum = ethers.BigNumber.from(gasLimit.toString())
+                gasLimitNum = ethers.BigNumber.from(txGas.toString())
                  .add(ethers.BigNumber.from(5000))
                  .toNumber();
 
-                _logMessage(`Gas limit calculated for method ${methodName} in SDK: ${gasLimitNum}`);
-                _logMessage("gas limit number" + gasLimitNum);
+                _logMessage(`Gas limit (txGas) calculated for method ${methodName} in SDK: ${gasLimitNum}`);
```
Comment (tomarsachin2271) [inline src/Biconomy.js:890]: gasLimitNum is logged twice with different wordings here. It should be logged once.

File: src/Biconomy.js (hunk 1 above)
Comment (livingrock7) [reply src/Biconomy.js:813]: ideally .gas should work. but I noticed when you send through ethers with txParam object, it doesnt recognise gas but gasLimit does. so I have added an OR

File: src/Biconomy.js (hunk 2 above)
Comment (livingrock7) [reply src/Biconomy.js:890]: done

File: src/Biconomy.js (hunk 1 above)
Comment (livingrock7) [reply src/Biconomy.js:813]: txGas is part of request object in executeEIP712() in Biconomy Forwarder
so gasLimit would be something user intended to pass for whole transaction
and txGas is req.txGas thats optional custom value

**Pull request merged**